src/repositories/PriceRepository.ts

```diff
@@ -1,41 +1,120 @@
 import {inject, injectable} from 'inversify';
+import {getModelForClass} from '@typegoose/typegoose';
+import dayjs from 'dayjs';
+import NodeCache from 'node-cache';
+
+import {Price} from '../models/Price';
+import {Pair} from '../types/Feed';
 import Settings from '../types/Settings';
 
-import PriceAggregator from '../services/PriceAggregator';
-import {Pair, PairWithFreshness} from '../types/Feed';
+export type SavePriceProps = {
+  source: string;
+  symbol: string;
+  value: number;
+  timestamp: Date;
+  expireAt?: Date;
+};
+
+export type LatestPriceProps = {
+  source: string;
+  symbol: string;
+  timestamp: {
+    from?: Date;
+    to?: Date;
+  };
+};
 
 @injectable()
-class PriceRepository {
-  @inject('Settings') settings!: Settings;
-  priceAggregator: PriceAggregator;
+export class PriceRepository {
+  cache: NodeCache;
+  defaultPriceTTL: number;
+
+  constructor(@inject('Settings') settings: Settings) {
+    this.cache = new NodeCache({stdTTL: 60, checkperiod: 60});
+    this.defaultPriceTTL = settings.mongodb.indexTTL.priceTTL;
+  }
+
+  async saveBatch(props: SavePriceProps[]): Promise<void> {
+    const operations = [];
+
+    for (const priceProp of props) {
+      const {source, symbol, value, timestamp} = priceProp;
+      const expireAt = priceProp.expireAt || dayjs().add(this.defaultPriceTTL, 'second').toDate();
+
+      operations.push({
+        updateOne: {
+          filter: {source, symbol, timestamp},
+          update: {source, symbol, timestamp, value, expireAt},
+          upsert: true,
+          new: true,
+        },
+      });
+    }
+
+    await getModelForClass(Price).bulkWrite(operations);
+  }
+
+  async getLatestPrice(props: LatestPriceProps): Promise<number | undefined> {
+    return (await this.getLatestPriceRecord(props))?.value;
+  }
+
+  async getLatestPriceRecord(props: LatestPriceProps): Promise<Price | undefined> {
+    const {
+      source,
+      symbol,
+      timestamp: {from, to},
+    } = props;
+
+    const timestampFilter: {[key: string]: Date} = {};
+    if (from) timestampFilter['$gte'] = from;
+    if (to) timestampFilter['$lt'] = to;
 
-  constructor(@inject(PriceAggregator) priceAggregator: PriceAggregator) {
-    this.priceAggregator = priceAggregator;
+    const price = await getModelForClass(Price)
+      .findOne(
+        {
+          source,
+          symbol,
+          timestamp: timestampFilter,
+        },
+        {},
+        {sort: {timestamp: -1}},
+      )
+      .exec();
+
+    return price || undefined;
   }
 
-  async savePrice(prefix: string, pair: string, price: number, timestamp: number): Promise<void> {
-    const formattedPair = pair.toUpperCase().replace('-', '~');
+  async getValueAndTimestamp(props: LatestPriceProps): Promise<{value: number; timestamp: number} | null> {
```

Comment:
```suggestion
  async getLatestPrice(props: LatestPriceProps): Promise<{value: number; timestamp: number} | null> {
```

Reply:
I didn't change this one because there is another one below. 
Please, let me know if I should change it

Reply:
another one was removed, right?